From 90d681ad626b465971f5a30c7f37c315eb543a5e Mon Sep 17 00:00:00 2001
From: Laur59 <148864407+Laur59@users.noreply.github.com>
Date: Sat, 3 Aug 2024 10:47:42 +0300
Subject: [PATCH] [Driver] Remove crt0.o in the baremetal link

Workaround waiting that link option -nostartfiles is taken into account.
Forcing crt0.o in the link brakes projects using CMake as extension for baremetal is .obj
---
 clang/lib/Driver/ToolChains/BareMetal.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/clang/lib/Driver/ToolChains/BareMetal.cpp b/clang/lib/Driver/ToolChains/BareMetal.cpp
index 102d12700229..852e0442f50a 100644
--- a/clang/lib/Driver/ToolChains/BareMetal.cpp
+++ b/clang/lib/Driver/ToolChains/BareMetal.cpp
@@ -487,11 +487,6 @@ void baremetal::Linker::ConstructJob(Compilation &C, const JobAction &JA,
     CmdArgs.push_back(Arch == llvm::Triple::aarch64_be ? "-EB" : "-EL");
   }
 
-  if (!Args.hasArg(options::OPT_nostdlib, options::OPT_nostartfiles,
-                   options::OPT_r)) {
-    CmdArgs.push_back(Args.MakeArgString(TC.GetFilePath("crt0.o")));
-  }
-
   Args.addAllArgs(CmdArgs, {options::OPT_L, options::OPT_T_Group,
                             options::OPT_s, options::OPT_t, options::OPT_r});
 
-- 
2.39.5 (Apple Git-154)

